docs/internals/ARCHITECTURE.md

```diff
@@ -103,127 +98,118 @@ hash map that stores accounts keyed by their IDs. Processing a transfer is then
+This is somewhat idiosyncratic use of the term so let's be precise about what TigerBeetle does and 
+does not do.
 
 Unlike some embedded systems, TigerBeetle doesn't use global statics (`.bss` section) for allocation.
 The memory usage of TigerBeetle depends on the CLI arguments passed at runtime.
 
-TigerBeetle also doesn't do arena allocation. One way you can ensure that an application uses at
-most, say, 4 GiB of memory is by allocating an appropriately fixed-sized arena at the start, and
-using this arena for all allocations, failing with out of memory error if more than 4 GiB has to be
-allocated.
-
-In TigerBeetle, static memory allocation is not a goal, but the consequence of the fact that
-everything has an explicit upper bound.
-
-When TigerBeetle starts, it looks at the CLI arguments and then computes, for every programming
-"object" in the system, how many such objects would be needed in the worst-case, and preallocates
-all of the objects. After startup, no new objects are created, and, as a consequence, no dynamic
-memory allocation is needed. The upper bounds are computed using a combination of static and dynamic
-information.
-
-Knowing the limits is useful, because it ensures that the system continues to function correctly
-even when overloaded. For example, TigerBeetle doesn't need to have _explicit_ code for handling
-backpressure --- if everything has a limit, there's nothing to grow without bound to begin with.
-Backpressure arises naturally as the property of the entire system from the pairwise interactions
-between components which need to honor each-other's limits.
-
-Here's another interesting consequence of static limits. One problem in highly concurrent
-applications is runaway concurrency --- spawning more concurrent tasks than the number of underlying
-resources available which leads to oversubscription. A concurrent task is usually a closure
-allocated somewhere on the heap and registered with an event loop, a `Box<dyn Future>`. Because
-TigerBeetle _can't_ heap allocate structures at runtime, it follows that each TigerBeetle "future"
-is represented by an explicit struct which is statically allocated as a field of a  component that
-owns the structure. So there's always a natural limit of how many concurrent tasks can be in flight!
-
-In summary, **static allocation is not the goal, but rather a forcing function to ensure that
-everything has a limit**, and a natural consequence of the limits.
+TigerBeetle also doesn't do arena allocation. It allocates an appropriately fixed-sized arena for 
+allocations at the start and fails with an 'out of memory' error if the limit is exceeded. In 
+TigerBeetle, static memory allocation is a consequence of everything having an explicit upper bound.
+
+When TigerBeetle starts, it computes the worst case memory allocation scenario for every "object" in 
+the system. After startup, no new objects are created therefore no dynamic memory allocation 
```

Comment:
>worst case memory allocation scenario for every "object" 

It's worth leaving the implication here: what TB actually computes is the _count_ of objects, and worst-case memory allocation is the result of the count. We actually can drop the memory part, and leave only the counts, as that's more important.